pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -532,34 +532,36 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
       properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
     }
 
-    String minValue = columnIndexCreationInfo.getMin().toString();
-    String maxValue = columnIndexCreationInfo.getMax().toString();
+    // NOTE: Min/max could be null for real-time aggregate metrics.
+    Object min = columnIndexCreationInfo.getMin();
+    Object max = columnIndexCreationInfo.getMax();
+    if (min != null && max != null) {
+      addColumnMinMaxValueInfo(properties, column, dataType, min.toString(), max.toString());
+    }
+
     String defaultNullValue = columnIndexCreationInfo.getDefaultNullValue().toString();
     if (dataType == DataType.STRING) {
       // Check special characters for STRING column
-      if (isValidPropertyValue(minValue)) {
-        properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
-      }
-      if (isValidPropertyValue(maxValue)) {
-        properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
-      }
       if (isValidPropertyValue(defaultNullValue)) {
         properties.setProperty(getKeyFor(column, DEFAULT_NULL_VALUE), defaultNullValue);
       }
     } else {
-      properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
-      properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
       properties.setProperty(getKeyFor(column, DEFAULT_NULL_VALUE), defaultNullValue);
     }
   }
 
-  public static void addColumnMinMaxValueInfo(PropertiesConfiguration properties, String column, String minValue,
-      String maxValue) {
-    // Check special characters for STRING column
-    if (isValidPropertyValue(minValue)) {
+  public static void addColumnMinMaxValueInfo(PropertiesConfiguration properties, String column, DataType dataType,
+      String minValue, String maxValue) {
+    if (dataType == DataType.STRING) {
```

Comment:
Why do we need to check for data type?

Reply:
To save the unnecessary checks for property value. But after reconsideration, since it is not in performance critical path, change it to always validate for simplicity and clarity.